src/App/Pages/Generator/GeneratorPageViewModel.cs

```diff
@@ -542,6 +546,64 @@ public bool ShowSimpleLoginApiKey
 
         public string ShowSimpleLoginHiddenValueIcon => _showSimpleLoginApiKey ? BitwardenIcons.EyeSlash : BitwardenIcons.Eye;
 
+        public string DuckDuckGoApiKey
+        {
+            get => _usernameOptions.DuckDuckGoApiKey;
+            set
+            {
+                if (_usernameOptions.DuckDuckGoApiKey != value)
+                {
+                    _usernameOptions.DuckDuckGoApiKey = value;
+                    TriggerPropertyChanged(nameof(DuckDuckGoApiKey));
+                    SaveUsernameOptionsAsync(false).FireAndForget();
+                }
+            }
+        }
+
+        public bool ShowDuckDuckGoApiKey
+        {
+            get
+            {
+                return _showDuckDuckGoApiKey;
+            }
+            set => SetProperty(ref _showDuckDuckGoApiKey, value,
+                additionalPropertyNames: new string[]
+                {
+                    nameof(ShowDuckDuckGoHiddenValueIcon)
+                });
+        }
+
+        public string ShowDuckDuckGoHiddenValueIcon => _showDuckDuckGoApiKey ? BitwardenIcons.EyeSlash : BitwardenIcons.Eye;
+
+        public string FastmailApiKey
+        {
+            get => _usernameOptions.FastMailApiKey;
+            set
+            {
+                if (_usernameOptions.FastMailApiKey != value)
+                {
+                    _usernameOptions.FastMailApiKey = value;
+                    TriggerPropertyChanged(nameof(FastmailApiKey));
+                    SaveUsernameOptionsAsync(false).FireAndForget();
+                }
+            }
+        }
+
+        public bool ShowFastmailApiKey
+        {
+            get
+            {
+                return _showFastmailApiKey;
+            }
+            set => SetProperty(ref _showFastmailApiKey, value,
+                additionalPropertyNames: new string[]
+                {
+                    nameof(ShowFastmailHiddenValueIcon)
+                });
+        }
+
+        public string ShowFastmailHiddenValueIcon => _showFastmailApiKey ? BitwardenIcons.EyeSlash : BitwardenIcons.Eye;
+
```

Comment:
If possible, refactor this alongside with the page xaml so we have common properties used that change their values depending on the forwarded email alias service selected.